buildengine/engine.go

```diff
@@ -223,13 +221,41 @@ func (e *Engine) Dev(ctx context.Context, period time.Duration) error {
 }
 
 func (e *Engine) buildAndDeploy(ctx context.Context, replicas int32, waitForDeployOnline bool, modules ...string) error {
-	err := e.buildWithCallback(ctx, func(ctx context.Context, module Module) error {
-		return Deploy(ctx, module, replicas, waitForDeployOnline, e.client)
-	}, modules...)
-	if err != nil {
-		return err
+	if len(modules) == 0 {
+		modules = maps.Keys(e.modules)
 	}
-	return nil
+
+	deployQueue := make(chan Module, len(modules))
+	wg, ctx := errgroup.WithContext(ctx)
+	wg.SetLimit(runtime.NumCPU())
```

Comment:
I tried to follow the pattern in the `build` command to limit the concurrency. Not 100% sure if this is the best solution of if there's something else I should try here.

Reply:
You probably don't need the same limit for deploys, it will be mostly network bound I think. You can probably remove the limit for now.